src/screens/InitWallet.js

```diff
@@ -235,9 +235,53 @@ class NewWordsScreen extends React.Component {
   }
 }
 
+/**
+ * Handles the validation of the raw seed words input
+ * @param {string} rawWords String containing all the words written by the user
+ * @returns {{words: string[], errorMessage: string, isValid: boolean}}
+ */
+function handleValidWords(rawWords) {
+  let trimmedWordsStr = rawWords.trim(/\s+/);
+  let errorMessage = '';
+  let isValid = false;
+
+  try {
+    const { valid, words } = walletUtils.wordsValid(trimmedWordsStr);
+    trimmedWordsStr = words;
+    isValid = valid;
+  } catch (e) {
+    // Handling unknown errors
+    if (!(e instanceof hathorErrors.InvalidWords)) {
+      throw e;
+    }
+
+    // Handling Invalid Words errors
+    isValid = false;
+    errorMessage = e.message;
+    const listOfInvalidWords = e.invalidWords.filter((word) => word.length !== 0);
```

Comment:
Is this filter necessary? I think the method will return only the invalid words. Or it could return an empty string?

Reply:
Yes, it could! That would result in an error message like `List of invalid words: ` , with no words at all.